src/coins.h

```diff
@@ -146,16 +164,36 @@ struct CCoinsCacheEntry
     inline void AddFlags(uint8_t flags, CoinsCachePair& self, CCoinsCacheEntry& sentinel) noexcept
     {
         assert(&self.second == this);
+        if (!m_flags && flags) {
```

Comment:
Also, `m_prev` and `m_next` would have to be `CoinsCachePair`s, not `CCoinsCacheEntry`s, so we can't set the pointers in the constructor since we don't have a pair.

Reply:
Done! Thank you for your review, I think it's pretty clean now. I just had to add a `SelfRef` method on `CCoinsCacheEntry` for the sentinel to init its pointers to itself.
Also, we need to not call `Next()` randomly on any `CCoinsCacheEntry`, only from the sentinel and then iterating through. This isn't done anywhere in the patch, but it's not protected against if anyone does. Now instead of returning a `nullptr` if it's not in the linked list it could potentially return a pointer to an already freed entry.

Reply:
Feel free to disregard: Would it make sense for `Next()`/`Prev()` to add an `Assume(m_flags)` or to do `if(m_flags) { return m_next; }` to protect against the dangling reference?

Reply:
Added `Assume(m_flags)`, thank you for the suggestion @davidgumberg! I did have to add `m_flags = DIRTY;` to the `SelfRef` method so that we can call `Next` on the sentinel to get the first entry.